### PR TITLE
Bitacross worker build and publish workflow

### PR DESCRIPTION
**Pull request by kziemianek**
Adds github actions workflow for building`bitacross-worker` in `HW` mode and pushing it to dockerhub.

### CODE REVIEW

File: .github/workflows/build_bitacross.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Build bitacross-worker
+
+on:
+  workflow_dispatch:
```
Comment (Kailai-Wang) [inline .github/workflows/build_bitacross.yml:4]: Do you want to have a tag/branch/commit hash as input parameter?

File: .github/workflows/build_bitacross.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            SGX_MODE=HW
+            ADDITIONAL_FEATURES_ARG=      
+            IMAGE_FOR_RELEASE=true
+
+      - name: Build worker
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: bitacross-worker/build.Dockerfile
+          tags: litentry/bitacross-worker:latest
```
Comment (Kailai-Wang) [inline .github/workflows/build_bitacross.yml]: I suggest that the docker tag is parameterized too

**Review by Kailai-Wang — APPROVED** [review]
Looks good - the comment is more of a flexibility improvement

File: .github/workflows/build_bitacross.yml (hunk 2 above)
Comment (kziemianek) [reply .github/workflows/build_bitacross.yml]: It'll use `input.ref` for both checking out the code and tagging image

Comment (Kailai-Wang) [reply .github/workflows/build_bitacross.yml]: tags should be `${{ github.event.inputs.ref || github.ref }}` instead of latest?

Comment (kziemianek) [reply .github/workflows/build_bitacross.yml]: I would like to test this workflow now but seems I cannot start it from GitHub.

File: .github/workflows/build_bitacross.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Build local builder
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: bitacross-worker/build.Dockerfile
+          tags: local-builder:${{ github.event.inputs.ref || github.ref }}
+          target: builder
+          build-args: |
+            WORKER_MODE_ARG=offchain-worker
+            SGX_MODE=HW
```
Comment (Traf333) [inline .github/workflows/build_bitacross.yml:34]: does GH actions run on machine with HW support? I thought with GHA we can only use software mode :thinking:

**Comment by kziemianek** [comment]
Closing for now as procedure for building with `SGX_PRODUCTION=1` is not determined yet.

**Pull request closed without merging**